### ISSUE TITLE
Animate tile movement, or prevent animating token movement

### ISSUE THREAD

**Issue by itamarcu** [grandseiken/foundryvtt-vehicles-and-mechanisms#2]
It would be nice if tiles (and drawings) could animate when they are moved/rotated with this mod, similar to how tokens animate. This would make vehicles look much nicer, because currently when a vehicle moves/rotates, it snaps into position while all the tokens on top of it slowly move into position.

Ideally, there could be a toggle in the settings (or vehicle settings), where ON makes tiles and tokens animated and OFF makes them snap/teleport to the destination and rotation.

A gif showing the current behavior (which is the worst of both worlds):

![turtle vehicle](https://user-images.githubusercontent.com/6516621/94484558-4e6f7d00-01e5-11eb-9441-3c0eee347f4f.gif)

**Comment by grandseiken** [comment]
Yeah agreed that if you move something with tokens a long way it doesn't look ideal right now.

Disabling token animation would be an easy short-term fix.

The issue with animating tiles is this isn't something foundry does by default AFAIK, so I'll have to look into whether it's possible to apply the token animation system to tiles as well.

**Comment by grandseiken** [comment]
So I actually tried not animating the tokens, and it actually doesn't look much better right now. The reason for this is that, I am currently moving the different entity types one at a time, to avoid the possibility of running into data loss due to https://gitlab.com/foundrynet/foundryvtt/-/issues/3455, so both tiles and tokens do not animate, but they still move at slightly different times.

So I think even the easy fix will need more investigation to determine whether it is safe to update tokens + tiles simultaneously, and improvements to the update batcher if so, and otherwise may not be easily solvable without introducing other problems until Foundry 0.8.

**Comment by itamarcu** [comment]
What about animating them with `CanvasAnimation.animateLinear`? (though this will possibly only animate them in the client that moved/rotated them)

**Comment by Merudo** [comment]
I'm having the same problem with my boats - the boat tiles teleports instantly to the target location, but the tokens slowly move to their new place:

![ship](https://user-images.githubusercontent.com/7371085/113710498-6569bd00-96a9-11eb-871c-ca93cb3a562c.gif)

Is there a workaround currently? I tried making the boat tile a token instant, but when if I click on it by accident all the other tokens on deck become obscured by the boat.
